lib/puma/client.rb

```diff
@@ -223,7 +223,7 @@ def setup_body
 
       te = @env[TRANSFER_ENCODING2]
 
-      if te == CHUNKED
+      if te && te.downcase == CHUNKED
```

Comment:
Actually, I have no idea if this is a performance sensitive path or not, but using `casecmp` [will be ever so slightly faster here](https://github.com/JuanitoFatas/fast-ruby#stringcasecmp-vs-stringdowncase---code) (and allocates one fewer object):

```ruby
CHUNKED.casecmp(te) == 0
```